go-controller/pkg/crd/userdefinednetwork/v1/cudn.go

```diff
@@ -77,3 +84,137 @@ type ClusterUserDefinedNetworkList struct {
+	// subnets is optional. When omitted OVN-Kubernetes won't assign IP address automatically.
+	// Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed.
+	// The format should match standard CIDR notation (for example, "10.128.0.0/16").
+	// This field must be omitted if `ipam.mode` is `Disabled`.
+	// When physicalNetworkName points to the OVS bridge mapping of a network that provides IPAM services
+	// (e.g., a DHCP server), ipam.mode should be set to Disabled. This turns off OVN-Kubernetes IPAM and avoids
+	// conflicts with the existing IPAM services on this localnet network.
+	//
+	// +optional
+	Subnets DualStackCIDRs `json:"subnets,omitempty"`
+
+	// excludeSubnets is a list of CIDRs to be removed from the specified CIDRs in `subnets`.
+	// The CIDRs in this list must be in range of at least one subnet specified in `subnets`.
+	// excludeSubnets is optional. When omitted no IP address is excluded and all IP addresses specified in `subnets`
+	// are subject to assignment.
+	// The format should match standard CIDR notation (for example, "10.128.0.0/16").
+	// This field must be omitted if `subnets` is unset or `ipam.mode` is `Disabled`.
+	// When `physicalNetworkName` points to OVS bridge mapping of a network with reserved IP addresses
+	// (which shouldn't be assigned by OVN-Kubernetes), the specified CIDRs will not be assigned. For example:
+	// Given: `subnets: "10.0.0.0/24"`, `excludeSubnets: "10.0.0.200/30", the following addresses will not be assigned
+	// to pods: `10.0.0.201`, `10.0.0.202`.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=25
+	ExcludeSubnets []CIDR `json:"excludeSubnets,omitempty"`
+
+	// ipam configurations for the network.
+	// ipam is optional. When omitted, `subnets` must be specified.
+	// When `ipam.mode` is `Disabled`, `subnets` must be omitted.
+	// `ipam.mode` controls how much of the IP configuration will be managed by OVN.
+	//    When `Enabled`, OVN-Kubernetes will apply IP configuration to the SDN infra and assign IPs from the selected
+	//    subnet to the pods.
+	//    When `Disabled`, OVN-Kubernetes only assigns MAC addresses, and provides layer2 communication, and enables users
+	//    to configure IP addresses on the pods.
+	// `ipam.lifecycle` controls IP addresses management lifecycle.
+	//    When set to 'Persistent', the assigned IP addresses will be persisted in `ipamclaims.k8s.cni.cncf.io` object.
+	// 	  Useful for VMs, IP address will be persistent after restarts and migrations. Supported when `ipam.mode` is `Enabled`.
+	//
+	// +optional
+	IPAM *IPAMConfig `json:"ipam,omitempty"`
+
+	// mtu is the maximum transmission unit for a network.
+	// mtu is optional. When omitted, the configured value in OVN-Kubernetes (defaults to 1500 for localnet topology)
+	// is used for the network.
+	// Minimum value for IPv4 subnet is 576, and for IPv6 subnet is 1280.
+	// Maximum value is 65536.
```

Comment:
65535

Reply:
Hi, sorry for going back and forward on this, the max value validation should be 65536.
This is how the MTU validation are for Layer2 and Layer3 typologies, and also OCP clusterNetwork https://github.com/openshift/api/blob/master/network/v1/types.go#L62-L67

Reply:
uhm ok. I'm still very perplexed about 65536 being our max, but let's be coherent with what's in openshift/api then.